src/karmen_frontend/cypress/integration/gcodes/adding_spec.js

```diff
@@ -0,0 +1,78 @@
+import { Chance } from "chance";
+const chance = new Chance();
+
+const attachFile = (file) => cy.get("input[name=file]").attachFile(file);
+const submitForm = () => cy.get('button[type="submit"]').click();
+
+const testGCodeIsAdded = (organizationUuid) => {
+  cy.location().then((loc) => {
+    expect(loc.pathname).to.contains(`/${organizationUuid}/gcodes`);
+  });
+};
+
+describe("G-codes: Adding", function () {
+  let email, password, organizationUuid;
+  beforeEach(() => {
+    email = chance.email();
+    password = chance.string();
+    return cy
+      .logout()
+      .createUser(email, password)
+      .login(email, password)
+      .then((data) => {
+        organizationUuid = Object.keys(data.organizations)[0];
+        cy.get('button[id="navigation-menu-toggle"]').click();
```

Comment:
Can't we use [testing lib for cypress](https://github.com/testing-library/cypress-testing-library) and move towards more semantic querying? Id-based queries are prone to changes and generally fragile as hell.

This is a general requirement that needs to be applied on multiple places.

Reply:
Done for the tests that I have added here, but there is still approx. more than 90 occurrences of `cy.get("...selector")` that probably needs to be rewrited. Are you requesting to refactor previous tests in this PR or we separate it to another issue/branch?

Reply:
Nope, let's fix that once it breaks (as it will).